android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/AuthManager.java

```diff
@@ -0,0 +1,630 @@
+        checkNotNull(signUp);
+        checkNotNull(signUp.getEmail());
+        checkNotNull(signUp.getPassword());
+
+        logger.debug("signUp called with signUp: " + signUp);
+
+        signUp.study(config.getStudyId())
+                .consent(false)
+                .status(null);
+
+        return RxUtils.toBodySingle(authenticationApi.signUp(signUp))
+                .doOnSuccess(message -> {
+                    accountDAO.setSignIn(
+                            new SignIn()
+                                    .study(config.getStudyId())
+                                    .email(signUp.getEmail())
+                                    .password(signUp.getPassword()));
+
+                    StudyParticipant participant = new StudyParticipant();
+                    participant.email(signUp.getEmail())
+                            .firstName(signUp.getFirstName())
+                            .lastName(signUp.getLastName())
+                            .externalId(signUp.getExternalId());
+
+                    accountDAO.setStudyParticipant(participant);
+                }).doOnError(throwable -> {
+                    accountDAO.setSignIn(null);
+                    accountDAO.setStudyParticipant(null);
+                }).toCompletable();
+    }
+
+    /**
+     * @param email participant's email address
+     * @return notifies of completion or error
+     */
+    @NonNull
+    public Completable resendEmailVerification(@NonNull final String email) {
+        checkNotNull(email);
+
+        logger.debug("resendEmailVerification called with email: " + email);
+
+        return RxUtils.toBodySingle(
+                authenticationApi.resendEmailVerification(
+                        new Email()
+                                .study(config.getStudyId())
+                                .email(email)))
+                .toCompletable();
+    }
+
+    /**
+     * On success, stores participant's email, password and session. If a
+     * NotAuthenticatedException is encountered, this could mean the credentials are invalid, or the
+     * user has not verified their email.
+     *
+     * @param email    participant's email address
+     * @param password participant's password
+     * @return bridge response
+     */
+    @NonNull
+    public Single<UserSessionInfo> signIn(@NonNull final String email, @NonNull final String
+            password) {
+        checkNotNull(email);
+        checkNotNull(password);
+
+        logger.debug("signIn called with email: " + email);
+
+        SignIn signIn = new SignIn()
+                .study(config.getStudyId())
+                .email(email)
+                .password(password);
+
+        // store sign in, so we have signIn as a key to retrieve session in case of 412
+        accountDAO.setSignIn(signIn);
+
+        return RxUtils.toBodySingle(
+                authenticationApi.signIn(signIn))
+                .toObservable()
+                .retryWhen(retrySignInForCOnsentOnce())
```

Comment:
also, can you add a comment above .retryWhen, that explains the conditions in which it is called?